### PR TITLE
[Techdocs] [Followup #15417] Renames mentions of mkDocs in favor of mkdocs 

### PR DESCRIPTION
**Pull request by aitherios**
## Hey, I just made a Pull Request!

The PR https://github.com/backstage/backstage/pull/15417 had a good comment from @kmatyukhin . This patch intends to isolate the suggestions unblocking the merge of https://github.com/backstage/backstage/pull/15417 .

The suggestion was naming consistency. This patch renames mentions of `mkDocs` in favor of `mkdocs`.

@kmatyukhin could you please take a look and help me out. I need to know what should I write on `Co-authored-by`.

#### :heavy_check_mark: Checklist

- [x] Change `Co-authored-by` to information provided by @kmatyukhin 
- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by kmatyukhin** [comment]
Oh, my contribution was so small to worth mentioning 😳. But if you ask I'm Konstantin Matyukhin <kmatyukhin@gmail.com>.

File: plugins/techdocs-node/api-report.md [hunk 1]
```diff
@@ -89,7 +89,7 @@ export const getLocationForEntity: (
 ) => ParsedLocationAnnotation;
 
 // @public
-export const getMkDocsYml: (
+export const getMkdocsYml: (
```
Comment (camilaibs) [inline plugins/techdocs-node/api-report.md:92]: I believe you have to deprecate this and export a new one, otherwise, it would be a breaking change.

Comment (aitherios) [reply plugins/techdocs-node/api-report.md:92]: Good catch. Thank you.

File: packages/techdocs-cli/cypress/e2e/mkdocs_serve.cy.js [hunk 2]
```diff
@@ -16,7 +16,7 @@
 /// <reference types="cypress" />
 describe('TechDocs Live Preview - MkDocs Serve', () => {
   it('successfully serves documentation', () => {
-    cy.visit(Cypress.env('mkDocsBaseUrl'));
```
Comment (aitherios) [inline packages/techdocs-cli/cypress/e2e/mkdocs_serve.cy.js:19]: @Rugvip & @camilaibs, I can't find the cypress.json file anymore. Where these environment variables are stored?

Comment (Rugvip) [reply packages/techdocs-cli/cypress/e2e/mkdocs_serve.cy.js:19]: No idea 😅

**Review by Rugvip — APPROVED** [review]
🟢 to unblock

**Review by camilaibs — COMMENTED** [review]
LGTM 🎉

**Review by agentbellnorm — APPROVED** [review]
Looks good to me!

**Review by camilaibs — APPROVED** [review]

**Pull request merged**